doc/tdshim_spec.md

```diff
@@ -762,6 +762,55 @@ The expected flow is:
    TDH.MR.EXTEND as described above, from the lowest GPA to the highest GPA,
    with 256 bytes granularity.
 
+#### Guideline
+
+TD Shim should follow [TCG Platform Firmware Profile specification](#tcg-pfp)
+in general.
+
+1. About separator
+
+   TD Shim shall measure 0x00000000 or 0xFFFFFFFF to RTMR[0~1] and the matching
+   EV_SEPARATOR event shall be recorded in the event log, prior to the
+   invocation of TD payload. As such, the verifier can know the rest events are
+   for the TD payload.
+
+   In case of an error occurs, TD shim shall measure 0x00000001 to RTMR[0~1] and
+   the matching EV_SEPARATOR event shall be recorded in the event log.
+
+2. About TD configuration
+
+   TD Shim shall extend the TD Hob to RTMR[0] and the event should be
+   EV_EFI_HANDOFF_TABLES2 with event data being UEFI_HANDOFF_TABLE_POINTERS2
+   and TableDescription being "td_hob".
+
+3. About payload info
+
+   TD Shim shall extend the payload binary to RTMR[1] and the event should be
+   EV_EFI_PLATFORM_FIRMWARE_BLOB2 with event data being
+   UEFI_PLATFORM_FIRMWARE_BLOB2 and BlobDescription being "td_payload".
+
+   TD Shim shall extend the payload parameter to RTMR[1] and the event should be
+   EV_EFI_PLATFORM_FIRMWARE_BLOB2 with event data being
+   UEFI_PLATFORM_FIRMWARE_BLOB2 and BlobDescription being "td_payload_parameter".
```

Comment:
Do we need to include the content of the payload parameter in the event data instead of just a descriptive information?

Reply:
Do you mean "td_payload_parameter:\<param_string\>" ?

According to TCG, the size of descriptor is 256 byte. I am not sure if it is big enough..
Is truncation in descriptor acceptable?

Reply:
What do you mean by truncation? Does this mean that if one event entry is not enough, use multiple entries to record the contents of the payload parameter?

Reply:
Sorry for the confusion. Truncation means only partial of the data is recorded. For example:
Let's assume the descriptor space is small, only allows 24 bytes.
While the boot parameter is very long "1234567890abcdef1234567890abcdef" - 32 bytes.
The hash will be HASH("1234567890abcdef1234567890abcdef") - no truncation.
But the descriptor will be "1234567890abcdef12345678" - it is truncated.

I am thinking below options:

A. Current UEFI_PLATFORM_FIRMWARE_BLOB2 includes a pointer and length for the data. But not data itself. that is why I thought to put parameter string in descriptor.
```
typedef struct tdUEFI_PLATFORM_FIRMWARE_BLOB2 {
    UINT8 BlobDescriptionSize;
    BYTE[BlobDescriptionSize] BlobDescription;
    UEFI_PHYSICAL_ADDRESS BlobBase;
    UINT64 BlobLength;
} UEFI_PLATFORM_FIRMWARE_BLOB2;
```

If you can put parameter string in td-shim image, then the verifier can already find the data by using pointer. No need to put parameter string in descriptor.

The flow is:
1. Verifier finds the "td_payload_parameter"
2. Verifier gets the UEFI_PLATFORM_FIRMWARE_BLOB2.BlobBase and BlobLength.
3. Verifier gets the corresponding parameter string there.
Does that resolve the problem?


If you really really want to include the data, then we have to use other type. For example:

B. EV_PLATFORM_CONFIG_FLAGS.
The event data could be the parameter string.

What do you think?

Reply:
I think scheme B may be more appropriate, because in normal verification scenarios, the verifier may not be able to get and load the td-shim image. Using scheme B allows the verifier to verify the parameter settings in a more flexible and lightweight way.

Reply:
I think scheme B may be more appropriate because if the verifier obtains and loads the referenced TD shim image and reads the parameters through the pointer in event data, if the parameters measured in the eventlog have been tampered with, the verifier cannot accurately know which parameters are not configured as expected.

Reply:
agree. Will update the proposal.